### ISSUE TITLE
Error `Cannot read property 'username' of undefined` after logout

### ISSUE THREAD

**Issue by cpxPratik** [Kinto/kinto-admin#577]
Signing with basic auth then logging out, then cannot sign back in. Show error as in the pic.

`docker-compose.yml`

```
db:
  environment:
    PGDATA: /var/lib/postgresql/data/pgdata
    POSTGRES_DB: postgres
    POSTGRES_PASSWORD: postgres
    POSTGRES_USER: postgres
  image: postgres
  volumes:
    - ./postgresql/data:/var/lib/postgresql/data
web:
  image: kinto/kinto-server
  links:
   - db
  ports:
   - "8848:8888"
  volumes:
    - ./config:/etc/kinto

```
`./config/kinto.ini`

```
[server:main]
use = egg:waitress#main
host = 0.0.0.0
port = %(http_port)s

[app:main]
use = egg:kinto

kinto.cache_backend = kinto.core.cache.postgresql
kinto.cache_url = postgres://postgres:postgres@db/postgres
kinto.storage_backend = kinto.core.storage.postgresql
kinto.storage_url = postgres://postgres:postgres@db/postgres
kinto.permission_backend = kinto.core.permission.postgresql
kinto.permission_url = postgres://postgres:postgres@db/postgres

kinto.includes = kinto.plugins.default_bucket
                 kinto.plugins.admin
#                kinto.plugins.accounts
#                kinto.plugins.history
#                kinto.plugins.quotas

```
![kintoadminerror](https://user-images.githubusercontent.com/6893334/43312221-541a3fca-91ac-11e8-9d3a-acdf2a7f4d1d.png)

If I open new tab and paste the URL then I can sign in from there.

**Comment by leplatrem** [comment]
Thanks for reporting!

The problem is probably around here, with the restored auth:
https://github.com/Kinto/kinto-admin/blob/51c44e041b75737689c75d24ae9e56ef078d6078/src/index.js#L37-L41

@magopian I remember you said something about logout, could this be related?

**Comment by magopian** [comment]
I'll investigate. The issue with the logout (since the integration of the openID connect method) is that it needs a refresh of the page.

@cpxPratik is it possible to login again after a refresh? If so I guess this is related to my comment in https://github.com/Kinto/kinto-admin/pull/509#issuecomment-389901873.

I thought i had opened an issue for that, but it seems I didn't. So thanks @cpxPratik for reporting it!

**Comment by cpxPratik** [comment]
@magopian No, it does not work after refresh. I am just using basic auth for now, not openID. Opening URL in new tab works though. While insecting storage, I can see only difference being that key `kinto-admin-server-history` is available with value `[{"server":"http://localhost:8848/v1/","authType":"basicauth"}]` on sessionstorage. When opening in new tab, there is no session storage values.

**Comment by cpxPratik** [comment]
Can confirm the issue is fixed with the 1.20 release.
